From 19d61962acdb3afa377631dc2d6658e8e73ac9ac Mon Sep 17 00:00:00 2001
From: Nir Soffer <nirsof@gmail.com>
Date: Mon, 25 Jun 2018 20:56:52 +0300
Subject: [PATCH] v2v: rhv-upload-plugin: Fix name error

"http" and "transfer" variables were missing in emulate_zero, so the
code would fail with NameError.  This can happen only when communicating
with old imageio versions not supporting the "zero" feature.

Testing with qemu-img 2.12 show that we never send emulated zero request
because of the highestwrite mechanism, but it can break with older
qemu-img-rhev used on RHEL.

(cherry picked from commit 0ae61ce99c351f9cda598016fb55ccc50313df67)
---
 v2v/rhv-upload-plugin.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/v2v/rhv-upload-plugin.py b/v2v/rhv-upload-plugin.py
index 2eec375f7..10887c031 100644
--- a/v2v/rhv-upload-plugin.py
+++ b/v2v/rhv-upload-plugin.py
@@ -338,6 +338,9 @@ def zero(h, count, offset, may_trim):
     r.read()
 
 def emulate_zero(h, count, offset):
+    http = h['http']
+    transfer = h['transfer']
+
     # qemu-img convert starts by trying to zero/trim the whole device.
     # Since we've just created a new disk it's safe to ignore these
     # requests as long as they are smaller than the highest write seen.
-- 
2.17.2

